src/cnaas_nms/devicehandler/underlay.py

```diff
@@ -12,7 +12,7 @@
 def find_free_infra_ip(session) -> Optional[IPv4Address]:
     """Returns first free IPv4 infra IP."""
     used_ips = []
-    device_query = session.query(Device).filter(Device.infra_ip is not None).options(load_only("infra_ip"))
+    device_query = session.query(Device).filter(Device.infra_ip != None).options(load_only("infra_ip"))  # noqa: E711
```

Comment:
Pretty weird that SQLAlchemy doesn't accept `is not None` in a filter - as this looks really ugly as Python code. But does it support `is None`? SQLAlchemy does support filter negation, doesn't it (not too familiar with SQLAlchemy)?  I'm thinking you could do something like `.exclude(Device.infra_ip is None)` instead, perhaps...

Reply:
They had an option to do Device.infra_ip.isnot(None) but I felt this looks less unique/special

Reply:
Working with NULL values in SQL isn't as straightforward as working with `None` in Python (e.g. NULL isn't equal to NULL in SQL), so this might actually be preferable.